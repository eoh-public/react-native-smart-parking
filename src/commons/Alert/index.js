import React from 'react';
import Modal from 'react-native-modal';
import { View, StyleSheet } from 'react-native';

import Text from '../Text';

import { Colors } from '../../configs';
import TextButton from '../TextButton';

class Alert extends React.Component {
  static _ref = null;

  static setRef(ref = {}) {
    this._ref = ref;
  }

  static getRef() {
    return this._ref;
  }

  static clearRef() {
    this._ref = null;
  }

  state = {
    title: '',
    description: '',
    buttons: [],
    show: false,
    styleButtons: [] || {},
  };

  static alert = (title, description, buttons, styleButtons) => {
    this._ref.setState({
      title,
      description,
      buttons,
      show: true,
      styleButtons,
    });
  };

  renderContent = () => {
    const { description, title } = this.state;
    return (
      <>
        <Text style={styles.title}>{title}</Text>
        <View style={styles.descriptionWrap}>
          {React.isValidElement(description) ? (
            description
          ) : (
            <Text style={styles.description}>{description}</Text>
          )}
        </View>
      </>
    );
  };

  renderButtons = () => {
    const justifyContent =
      this.state.buttons.length === 1 ? 'center' : 'space-between';
    return (
      <View
        key={this.state.title}
        style={[
          styles.flexRow,
          {
            justifyContent: justifyContent,
          },
          this.state.styleButtons,
        ]}
      >
        {this.state.buttons.map((item, index) => {
          return (
            <TextButton
              key={`item_${index}`}
              primary={item.primary}
              title={item.label}
              onPress={item.onPress}
              wrapStyle={item.style}
            />
          );
        })}
      </View>
    );
  };

  static hideModal = () => {
    this._ref.setState({ show: false });
  };

  static showModal = () => {
    this._ref.setState({ show: true });
  };

  render() {
    const { buttons, show } = this.state;
    return (
      <Modal animationIn={'zoomIn'} animationOut={'zoomOut'} isVisible={show}>
        <View style={[styles.wrap, styles.contentPadding]}>
          {this.renderContent()}
        </View>
        <View style={[styles.wrap, styles.buttonsPadding]}>
          {!!buttons?.length && this.renderButtons()}
        </View>
      </Modal>
    );
  }
}

const styles = StyleSheet.create({
  wrap: {
    padding: 20,
    alignItems: 'center',
    backgroundColor: Colors.White,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 18,
    color: Colors.Primary,
  },
  descriptionWrap: {
    minHeight: 78,
    paddingVertical: 10,
  },
  description: {
    textAlign: 'center',
  },
  flexRow: {
    flexDirection: 'row',
  },
  contentPadding: {
    paddingBottom: 0,
  },
  buttonsPadding: {
    paddingTop: 0,
    alignItems: 'stretch',
  },
});

export default Alert;
