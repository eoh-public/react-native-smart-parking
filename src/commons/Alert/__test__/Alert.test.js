import React from 'react';
import renderer, { act } from 'react-test-renderer';
import Alert from '../index';

describe('Test Alert', () => {
  let tree;
  test('create Alert', () => {
    act(() => {
      tree = renderer.create(<Alert />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
