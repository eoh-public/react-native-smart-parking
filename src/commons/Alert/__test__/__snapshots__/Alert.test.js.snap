// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Alert create Alert 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "padding": 20,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#00979D",
              "fontSize": 18,
              "fontWeight": null,
            },
          ]
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "minHeight": 78,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": undefined,
                "lineHeight": undefined,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "padding": 20,
          },
          Object {
            "alignItems": "stretch",
            "paddingTop": 0,
          },
        ]
      }
    />
  </View>
</Modal>
`;
